tls/s2n_x509_validator.c

```diff
@@ -368,81 +377,70 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_chain(struct s2n_x509_
         time_t current_time = (time_t)(current_sys_time / 1000000000);
         X509_STORE_CTX_set_time(validator->store_ctx, 0, current_time);
 
-        op_code = X509_verify_cert(validator->store_ctx);
+        int verify_ret = X509_verify_cert(validator->store_ctx);
+        if (verify_ret <= 0) {
+            int ossl_error = X509_STORE_CTX_get_error(validator->store_ctx);
+            switch (ossl_error) {
```

Comment:
This is a really nice fix! I appreciate that we can handle more errors here later if we want, making these failures a little less opaque :)